### ISSUE TITLE
mustermann: Mustermann::Sinatra#equal? returns incorrectly false for identical objects

### ISSUE THREAD

**Issue by deepj** [oracle/truffleruby#1958]
To reproduce:

```
gem install mustermann
ruby -rmustermann/sinatra -e "p Mustermann::Sinatra.new('/foo').equal?(Mustermann::Sinatra.new('/foo'))"
```
TruffleRuby: `false`
MRI: `true`

**Comment by bjfish** [comment]
Related to #1385 

The same specs fail in MRI with this change:
 ```
 class EqualityMap
    attr_reader :map

    def self.new
      # defined?(ObjectSpace::WeakMap) ? super : {}
      {}
    end
```

**Comment by norswap** [comment]
Despite 6ff7b590b66956194ec2a071643fa48e0e65507f, the test in the top comment still returns `false`, so this must be investigated further.

**Comment by norswap** [comment]
Fixed by 6ff7b590b66956194ec2a071643fa48e0e65507f
